### DISCUSSION TITLE
Should auto-generated source files include "// <auto-generated>" so they are ignored by code analysis?

### DISCUSSION THREAD

**Discussion by JustinC-IADS** [CommunityToolkit/Maui#524] [Q&A] [answered]
I'm currently trying to enable code analysis on a dotnet Maui project, but when I do, I immediately get analysis hits from the CommunityToolkit generated code.

After some searching, I found the following discussion about auto-generated file being ignored.
https://github.com/dotnet/roslyn/issues/36619

![image](https://user-images.githubusercontent.com/100383542/181386660-f1789979-6069-4433-bd12-36e58834448a.png)

Is anyone else running into this? Is there another was to configure the analyzers to ignore these files?

**Comment by TheCodeTraveler** [comment]
Good suggestion! 

@pictos - what do you think?

**Reply by pictos** [reply]
Sounds like a good idea. I'll give it a try

**Reply by JustinC-IADS** [reply]
Awesome. Thanks for the quick replies.

**Comment by pictos** [comment, marked as answer]
@JustinC-IADS are you able to create a small project that I can use as reference to fix it?

**Reply by JustinC-IADS** [reply]
Yes. I'm uploading it. Let me know if you prefer a different way to receive it.

Here's a screenshot of the code analysis options I have at the moment (and I have turned on TreatWarningsAsErrors in the .csproj) 
![image](https://user-images.githubusercontent.com/100383542/181772488-2ace1343-3bb3-43cf-a2fd-9fc11e2dce66.png)

[CodeAnalysis_GeneratedCode_Sample.zip](https://github.com/CommunityToolkit/Maui/files/9220396/CodeAnalysis_GeneratedCode_Sample.zip)

Let me know if I can do anything else to help.

**Reply by JustinC-IADS** [reply]
Code analysis is also complaining about other things from the default Maui project template, but this example should demonstrate the undesired analysis of the generated source files.

**Reply by pictos** [reply]
Thanks @JustinC-IADS I'll take a look on it ASAP

**Reply by VladislavAntonyuk** [reply]
@pictos if I am not mistaken you fixed it

**Reply by pictos** [reply]
Yes, it's just not released yet. Will be in the vNext.

**Comment by CliffAgius** [comment]
Closed as answered
